scs2-session-visualizer-jfx/src/main/java/us/ihmc/scs2/sessionVisualizer/jfx/controllers/yoPieChart/YoPieChartInputController.java

```diff
@@ -0,0 +1,302 @@
+
+      if (success)
+         setSelectionHighlight(false);
+   }
+
+   private void handleMousePressed(MouseEvent event)
+   {
+      if (event.getButton() == MouseButton.PRIMARY)
+      {
+         hideContextMenu();
+      }
+   }
+
+   private void handleMouseReleased(MouseEvent event)
+   {
+      if (event.getButton() == MouseButton.PRIMARY)
+      {
+         if (yoVariableDropLabel != null && event.isStillSincePress())
+         {
+            messager.submitMessage(yoCompositeSelectedTopic, Arrays.asList(YoCompositeTools.YO_VARIABLE, yoVariableDropLabel.getText()));
+         }
+      }
+      else if (event.getButton() == MouseButton.SECONDARY)
+      {
+         if (yoVariableDropLabel != null && event.isStillSincePress())
+         {
+            ContextMenu contextMenu = newGraphContextMenu();
+            if (!contextMenu.getItems().isEmpty())
+            {
+               contextMenuProperty.set(contextMenu);
+               contextMenu.show(rootPane, event.getScreenX(), event.getScreenY());
+            }
+            event.consume();
+         }
+      }
+      else if (event.getButton() == MouseButton.MIDDLE)
+      {
+         if (yoCompositeSelected.get() != null)
+         {
+            String type = yoCompositeSelected.get().get(0);
+            if (type.equals(YoCompositeTools.YO_VARIABLE))
+            {
+               String fullname = yoCompositeSelected.get().get(1);
+               YoComposite yoComposite = yoCompositeSearchManager.getYoComposite(type, fullname);
+
+               if (yoComposite != null && filter.test(yoComposite.getYoComponents().get(0)))
+               {
+                  setYoVariableInput(yoComposite.getYoComponents().get(0));
+                  messager.submitMessage(yoCompositeSelectedTopic, null);
+               }
+            }
+         }
+      }
+   }
+
+   private void hideContextMenu()
+   {
+      if (contextMenuProperty.get() != null)
+         contextMenuProperty.set(null);
+   }
+
+   private ContextMenu newGraphContextMenu()
+   {
+      if (yoVariableDropLabel == null)
+         return null;
+
+      ContextMenu contextMenu = new ContextMenu();
+      MenuItem menuItem = new MenuItem("Remove " + yoVariableDropLabel.getText());
+      menuItem.setMnemonicParsing(false);
+      menuItem.setOnAction(e ->
+                           {
+                              for (PieChart.Data data : pieChart.getData())
+                              {
+                                 if (data.getName().equals(yoVariableDropLabel.getText()))
+                                 {
+                                    pieChart.getData().remove(data);
+                                    break;
+                                 }
+                              }
+                              setYoVariableInput(null);
+                           });
+      contextMenu.getItems().add(menuItem);
+      return contextMenu;
+   }
+
+   public void setYoVariableInput(YoVariable yoVariable)
```

Comment:
When a YoVariable is dropped into the grid point, we have logic to check if it already, exists, or to set the text to that variable so we can add it to the pie chart